AWSClientRuntime/Sources/Regions/DefaultRegionResolver.swift

```diff
@@ -5,26 +5,32 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 import AwsCommonRuntimeKit
+import ClientRuntime
 
 public struct DefaultRegionResolver: RegionResolver {
     public let providers: [RegionProvider]
-    
+    let logger: SwiftLogger
+
     public init(providers: [RegionProvider] = [EnvironmentRegionProvider(), ProfileRegionProvider(), IMDSRegionProvider()]) {
         // TODO: add more region resolvers i.e. System Properties, etc
         self.providers = providers
+        self.logger = SwiftLogger(label: "DefaultRegionProvider")
     }
     
     public func resolveRegion() -> String? {
         for provider in providers {
             do {
                 // TODO: Verify that this does not block calling thread. If it does, we need to solve this similar to the way we solve this for the credentials provider
+                logger.debug("Attempting to resolve region with: \(String(describing: type(of: provider)))")
                 if let region = try provider.resolveRegion().get() {
+                    logger.debug("Resolved region with: \(String(describing: type(of: provider)))")
                     return region
                 }
             } catch {
                 return nil
             }
         }
+        logger.debug("Unable to resolve region")
```

Comment:
can we update this to "Unable to resolve region using the DefaultRegionResolver chain" possibly?

Reply:
We are already outputting:

```
2021-11-03T14:55:14-0700 debug DefaultRegionProvider : Unable to resolve region
```

If we change it, it will look like:
```
2021-11-03T14:55:14-0700 debug DefaultRegionProvider : Unable to resolve region using the DefaultRegionResolver chain
```

This seems like duplicated information.  I'm assuming that we do not want to change this, but if we do, we can change it in a subsequent PR.

Reply:
Ah yes sorry I didn't realize that's how it looked. Glad you kept it the same